packages/@ember/-internals/runtime/lib/system/core_object.js

```diff
@@ -64,6 +66,21 @@ function initialize(obj, properties) {
       !(properties instanceof Mixin)
     );
 
+    let injectedProperties;
+    if (DEBUG) {
+      // these are all the implicit injectinos
+      injectedProperties = [];
+
+      let factory = getFactoryFor(obj);
+      if (factory) {
+        for (let injection in factory.injections) {
+          if (factory.injections[injection] === properties[injection]) {
```

Comment:
What is this checking? What is `properties` here?

Reply:
`properties` is `Object.assign({}, injections, createProps)` basically. So it's possible that the implicit injection has been overridden by an explicit value (an argument in a classic component for instance)